core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java

```diff
@@ -210,4 +218,18 @@ private Map<String, Object> getTaskResultsIfConfigurationAllows(Task task) {
         }
     }
 
+    /**
+     * Copy the headers parameter and add the Authorization header to it
+     *
+     * @param headers original list of headers
+     * @return new list containing the original headers + the authorization header
+     */
+    private List<Header> addAuthenticatedHeaderToHeaders(List<Header> headers) {
```

Comment:
what if the Authorization header is already present? should we overwrite it or prefer it?